File: rb/features.rb
# RealTimeBusData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module RealTimeBusDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      RealTimeBusDataBaseFeature.new
    when "test"
      RealTimeBusDataTestFeature.new
    else
      RealTimeBusDataBaseFeature.new
    end
  end
end

File: py/realtimebusdata_sdk/features.py
# RealTimeBusData SDK feature factory

from realtimebusdata_sdk.feature.base_feature import RealTimeBusDataBaseFeature
from realtimebusdata_sdk.feature.test_feature import RealTimeBusDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RealTimeBusDataBaseFeature(),
        "test": lambda: RealTimeBusDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
